graylog2-web-interface/src/components/search/Decorator.jsx

```diff
@@ -33,18 +33,40 @@ const Decorator = React.createClass({
     this.refs.editForm.open();
   },
   _handleSubmit(data) {
-    DecoratorsActions.update(this.props.decorator._id, { type: data.type, config: data.configuration });
+    DecoratorsActions.update(this.props.decorator._id, {
+      type: data.type,
+      config: data.configuration,
+      order: this.props.decorator.order,
+    });
   },
   _decoratorTypeNotPresent() {
     return {
-      name: <strong>Unknown decorator type</strong>,
+      name: 'Unknown decorator type',
     };
   },
+  // Attempts to resolve ID values in the decorator configuration against the type definition.
+  // This allows users to see actual names for entities in drop-downs, instead of their IDs.
+  _resolveConfigurationIds(config) {
+    const typeConfig = this.props.typeDefinition.requested_configuration;
+    const resolvedConfig = {};
+    const configKeys = Object.keys(config);
+
+    configKeys.forEach(key => {
+      const configValues = (typeConfig[key] ? typeConfig[key].additional_info.values : undefined);
+      if (typeof configValues === 'object' && !Array.isArray(configValues)) {
```

Comment:
I was being too cautious in case some other configuration field was using `values` for something else. I'll change the code 👍